src/js/components/FieldWrapper/FieldWrapper.stories.tsx

```diff
@@ -63,7 +65,15 @@ stories.add("Checkbox", () => {
       field="checkbox"
       status={status}
       message={status === "error" ? errorMessage : null}
-      fieldProps={{ options: [{ label: "Check me" }, { label: "Check me" }] }}
+      fieldProps={{
+        inputProps: {
+          onChange: (e: ChangeEvent<HTMLInputElement>) => {
+            console.log(e.target.value + ' clicked');
+            action(e.target.value + ' clicked');
+          },
```

Comment:
The reason your action isn't firing as expected is because the `action` function returns an event handler ([at least, in the version that we're currently on](https://github.com/storybookjs/storybook/tree/aeeafda2de3e97cd91e36c3dc7bdb0543d073429/addons/actions)). Ideally, the `action`'s responsne would be used in place of the onChange handler, but I like what you're doing with trying to differentiate. You should be able to do something like this instead:
```suggestion
          onChange: (e: ChangeEvent<HTMLInputElement>) => {
            action(e.target.value + ' clicked')(e);
          },
```
It might look a little weird, but you're just passing the synthetic event right to the action. I tested this locally and it worked for me :)

Reply:
Oh right, I totally forgot about that.